posthog-server/src/main/java/com/posthog/server/PostHogFeatureFlagOptions.kt

```diff
@@ -0,0 +1,119 @@
+package com.posthog.server
+
+/**
+ * Provides an ergonomic interface when providing options for capturing events
+ * This is mainly meant to be used from Java, as Kotlin can use named parameters.
+ */
+public class PostHogFeatureFlagOptions private constructor(
+    public val defaultValue: Any?,
+    public val groups: Map<String, String>?,
+    public val personProperties: Map<String, String>?,
+    public val groupProperties: Map<String, String>?,
+) {
+    public class Builder {
+        public var defaultValue: Any? = null
+        public var groups: MutableMap<String, String>? = null
+        public var personProperties: MutableMap<String, String>? = null
+        public var groupProperties: MutableMap<String, String>? = null
+
+        /**
+         * Sets the default value to return if the feature flag is not found or not enabled
+         */
+        public fun defaultValue(defaultValue: Any?): Builder {
+            this.defaultValue = defaultValue
+            return this
+        }
+
+        /**
+         * Add a single custom property to the capture options
+         */
+        public fun group(
+            key: String,
+            value: String,
+        ): Builder {
+            if (groups == null) {
+                groups = mutableMapOf()
+            }
+            groups!![key] = value
```

Comment:
not a big fan of `!!` but since its a builder, i dont see this being a problem (like concurrency issues)
maybe consider declaring the maps already created and later check if they are not empty instead?